als-suggestions/shared/src/main/scala/org/mulesoft/als/suggestions/plugins/aml/pathnavigation/DeclarablePathSuggestor.scala

```diff
@@ -0,0 +1,35 @@
+package org.mulesoft.als.suggestions.plugins.aml.pathnavigation
+
+import amf.core.client.scala.model.document.DeclaresModel
+import amf.core.client.scala.model.domain.NamedDomainElement
+import amf.shapes.client.scala.model.document.JsonSchemaDocument
+import org.mulesoft.als.suggestions.RawSuggestion
+
+import scala.concurrent.Future
+
+object DeclarablePathSuggestor {
+  def apply(schema: DeclaresModel, prefix: String): DeclarablePathSuggestor =
+    schema match {
+      case json: JsonSchemaDocument => JsonSchemaSuggestor(json, prefix)
+      case _                        => new DeclarablePathSuggestor(schema, prefix)
+    }
+
+}
+
+sealed class DeclarablePathSuggestor(schema: DeclaresModel, prefix: String) extends PathSuggestor {
+  override def suggest(): Future[Seq[RawSuggestion]] = {
+    val names = schema.declares.flatMap {
+      case named: NamedDomainElement => named.name.option()
+      case _                         => None
+    }
+    Future.successful(buildSuggestions(names.map(buildText), prefix))
+  }
+
+  def buildText(name: String): String = name
+}
+
+sealed case class JsonSchemaSuggestor(schema: JsonSchemaDocument, prefix: String)
+    extends DeclarablePathSuggestor(schema, prefix) {
+  override def buildText(name: String): String =
+    s"definitions/$name"
```

Comment:
Is this fix? does not take in consideration target document declarations path?
Take in mind that draft 19 support and recommends to use "$def" key for declarations. (this key will be store at an annotation on the document)

Reply:
at this point we don't have the target dialect available (it does not exist yet, but even so it is not avaliable in this context)
an alternative would be to use the sourceAST of the declaration and create a YPartBranch in order to print the complete path to the declaration, my intention was to avoid this as it would appear costly, but it could be an option

Reply:
I think the dialect/declarations path is also of no help in the case of draft19 (if AMF allows both `definitions` and `def` for declarations
if only 1 is allowed then there could be another `DeclarablePathSuggestor` with a different `buildText` that matches with this specific draft before

Reply:
Check the draft is not enough. You could check the draft if not any declaration is present (not makes so much sense because is nothing to suggest), but if there is already one declaration, for instance, "definitions", then you cannot suggest a reference to something like "$defs/Person" because would be invalid, no matter which draft version is. That why I think we should check first the if the declaration key annotation is present at the document (check what is wrote at the document).